Sources/PackageCollectionsSigning/Certificate/CertificatePolicy.swift

```diff
@@ -82,19 +86,295 @@ extension CertificatePolicy {
+                     callback: @escaping (Result<Void, Error>) -> Void) {
+        let wrappedCallback: (Result<Void, Error>) -> Void = { result in callbackQueue.async { callback(result) } }
+
+        let ocspURLs = CCryptoBoringSSL_X509_get1_ocsp(certificate.underlying)
+        defer { CCryptoBoringSSL_sk_OPENSSL_STRING_free(ocspURLs) }
+
+        let ocspURLCount = CCryptoBoringSSL_sk_OPENSSL_STRING_num(ocspURLs)
+        // Nothing to do if no OCSP URLs. Use `supportsOCSP` to require OCSP support if needed.
+        guard ocspURLCount > 0 else { return wrappedCallback(.success(())) }
+
+        // Construct the OCSP request
+        let digest = CCryptoBoringSSL_EVP_sha1()
+        guard let certid = OCSP_cert_to_id(digest, certificate.underlying, issuer.underlying),
+            let request = OCSP_REQUEST_new(),
+            OCSP_request_add0_id(request, certid) != nil else {
+            return wrappedCallback(.failure(CertificatePolicyError.ocspSetupFailure))
+        }
+        defer { OCSP_REQUEST_free(request) }
+
+        // Write the request binary to memory bio
+        guard let bio = CCryptoBoringSSL_BIO_new(CCryptoBoringSSL_BIO_s_mem()),
+            i2d_OCSP_REQUEST_bio(bio, request) > 0 else {
+            return wrappedCallback(.failure(CertificatePolicyError.ocspSetupFailure))
+        }
+        defer { CCryptoBoringSSL_BIO_free(bio) }
+
+        // Copy from bio to byte array then convert to Data
+        var count = 0
+        let out = UnsafeMutablePointer<UnsafePointer<UInt8>?>.allocate(capacity: 1)
+        guard CCryptoBoringSSL_BIO_mem_contents(bio, out, &count) > 0 else {
+            return wrappedCallback(.failure(CertificatePolicyError.ocspSetupFailure))
+        }
+
+        let requestData = Data(UnsafeBufferPointer(start: out.pointee, count: count))
+
+        let results = ThreadSafeArrayStore<Result<Bool, Error>>()
+        let group = DispatchGroup()
+
+        // Query each OCSP responder and record result
+        for index in 0 ..< ocspURLCount {
+            guard let urlStr = CCryptoBoringSSL_sk_OPENSSL_STRING_value(ocspURLs, numericCast(index)),
+                let url = String(validatingUTF8: urlStr).flatMap({ URL(string: $0) }) else {
+                results.append(.failure(OCSPError.badURL))
+                continue
+            }
+
+            let cacheKey = CacheKey(url: url, request: requestData)
+            if let cachedResult = self.resultCache[cacheKey] {
+                if cachedResult.timestamp + self.cacheTTL > DispatchTime.now() {
+                    results.append(.success(cachedResult.isCertGood))
+                    continue
+                } else {
+                    _ = self.resultCache.removeValue(forKey: cacheKey)
+                }
+            }
+
+            var headers = HTTPClientHeaders()
+            headers.add(name: "Content-Type", value: "application/ocsp-request")
+            url.host.map { headers.add(name: "Host", value: "\($0)") }
```

Comment:
`Host` is unconditional, and the URL request will fail if it's missing: a force-unwrap is probably more appropriate here.